testing/e2e/app/routes/Menu/spec.ts

```diff
@@ -0,0 +1,95 @@
+import { test, expect } from '@playwright/test';
+import { keyboardPressOptions } from '~/utils/utils.js';
+
+test.describe('Menu', () => {
+  test('should render menu items', async ({ page }) => {
+    await page.goto('/Menu');
+
+    expect(page.locator('.Menu')).toBeVisible();
+    for (let i = 0; i < 3; i++) {
+      expect(page.getByTestId(`MenuItem-${i}`)).toBeVisible();
+    }
+
+    await page.waitForTimeout(50);
+  });
+
+  test('should automatically handle conditional rendering', async ({
+    page,
+  }) => {
+    await page.goto('/Menu?visible=false');
+    await page.waitForTimeout(50);
+
+    expect(page.locator('.Menu')).toBeHidden();
+    for (let i = 0; i < 3; i++) {
+      expect(page.getByTestId(`MenuItem-${i}`)).toBeHidden();
+    }
+
+    const button = page.getByTestId('toggle-menu-button');
+    await button.click();
+
+    await expect(page.locator('.Menu')).toBeVisible();
+    for (let i = 0; i < 3; i++) {
+      await expect(page.getByTestId(`MenuItem-${i}`)).toBeVisible();
+    }
+
+    await page.waitForTimeout(50);
+  });
+
+  test('should render menu with custom role', async ({ page }) => {
+    await page.goto('/Menu?role=listbox');
+
+    expect(page.locator('.Menu[role="listbox"]')).toBeVisible();
+
+    await page.waitForTimeout(50);
+  });
+
+  test('should focus selected item', async ({ page }) => {
+    // TODO: Do we need to focus the selected item?
```

Comment:
Wanted to highlight this TODO, since I don't see it mentioned in the PR description.

Reply:
#2034 and #2036 removed uses of `Menu` from `ComboBox` and `Select`. As a result, focusing the selected item is no longer needed by any component that uses `Menu`.

Thus, I removed the code to focus the selected item and also removed this e2e test case.